src/Security/Authenticator.php

```diff
@@ -117,9 +117,10 @@ public static function register_authenticator($authenticator)
             return false;
         }
 
-        if (in_array($authenticator, self::$authenticators) == false) {
-            if (call_user_func(array($authenticator, 'on_register')) === true) {
-                array_push(self::$authenticators, $authenticator);
+        $authenticators = self::config()->get('authenticators');
+        if (in_array($authenticator, $authenticators) == false) {
+            if ($authenticator::on_register() === true) {
+                self::config()->merge('authenticators', [$authenticator]);
```

Comment:
Right, yep, that makes sense.

Leave the old "set/unset" methods in with a 4.0 deprecation notice, and update the docs to match the new Config.

Note to self: Add a check if the returned Authenticator in get_authenticators is actually a subclass of Authenticator.